feat: Quality tasks and quality action

johanseto opened the pull request:
## Description

The PR purpose is to integrate quality tasks into the repo.


## Additional information
This PR is based on a fork made in my profile. 
The initial  conversation is inthe PR: https://github.com/eduNEXT/codejailservice/pull/3
## Checklist for Merge

- [x] Update makefile
- [x] Add lint or quality action.
- [x] Clean python code
- [ ] Rebased master/main
- [ ] Squashed commits

<!--
You can put NA in the checklist if it doesn't apply

- [ X ] Check that don't apply / NA
-->

johanseto committed:
feat: Add license file

Create license file `LICENSE` with the terms of the AGPLv3.

johanseto committed:
clean: fix whitespace in pyfiles.

Fix some whitespace in `codejailservice/*`. Some pep 8 issues.

johanseto committed:
refactor: refactor `Makefile` with more options.

Refactor  file `Makefile` with common options used in other projects
like stackbuilder.

Also change and add common requirements in requirements folder.

johanseto committed:
style: Update py files to pass some quality tasks.

Add `setup.cfg` with pycodestyle,isort config.

Add `pylintrc` for pylint run configuration.

Add the doctring for pyfiles.

In `app.py` change logger to LOG var for good practices.

Add clean option in makefile.

johanseto committed:
feat: github action to run quality tasks.

Add github-action called `test.yml` in
github/workflows/

johanseto committed:
fix: fix configuration for import codejail.

Add configuration in `pylinttrc` and `setup.cfg` to recognize
codejail as a thirdparty lib.

johanseto committed:
docs: Apply suggestions in .py docstrings

Apply suggestions from code review in docstrings py files.

Co-authored-by: Felipe Montoya <felipe.montoya@edunext.co>

johanseto committed:
fix: solve issues of code review.

Update github action `quality.yml` for check all branches.

Check EOF in `pylintrc`, `setup.cfg`.

Quit test  section for individual PR.

Quit license for individual PR.

Quit wheel building and related with `setup.py` for individual PR.

johanseto committed:
Merge pull request #4 from johanv26/quality-tasks

Merge from fork in Johan-quality tasks

johanseto committed:
fix: solve bug in import order codejailsafe funcs.

Fix import lib of codejail safe functions according to pylint
code good practices and code execution.

Use importlib for lazy imports for codejail configuration.

Clean code according this change.

Delete `tutor.py` cause not changes used.

johanseto commented:
I test the changes in my local environment at this point and codejailservice works. The local container was configured with help of @ericfab179 using this service. 
Also, I remove `codejailservice/tutor.py` because my container doesn't present any error or change.

Next: 
- Test in other environments.
- Setup PR and test PR.

felipemontoya commented on Makefile:
```diff
@@ -1,7 +1,53 @@
+	pip install -r requirements/pip-tools.txt
+	pip-sync requirements/dev.txt
+
+
```
Remove this empty lines

johanseto replied:
Solved in 2a4a46f

felipemontoya commented on Makefile:
```diff
@@ -1,7 +1,53 @@
+	rm -fr dist/
+	rm -fr *.egg-info
+
+
```
extra empty line

johanseto replied:
Solved in 2a4a46f

felipemontoya commented on Makefile:
```diff
@@ -1,7 +1,53 @@
+.PHONY: quality  help requirements\
+	piptools selfcheck  upgrade test clean
```
typo: two spaces

johanseto replied:
Solved in 2a4a46f

felipemontoya commented on codejailservice/app.py:
```diff
@@ -56,22 +54,41 @@ def configure_codejail(app):
             )
 
+
+# -----------------app configuration-------------------------
```
This long kind of messages have a special syntax. See:  https://github.com/edx/edx-platform/blob/master/lms/envs/common.py#L57

johanseto replied:
Yes, I changed taking care the django syntax `######### FEAT #############`. Also I have a codestyle issue E266 for too many `#`. 
![image](https://user-images.githubusercontent.com/51926076/141316974-c96a935e-9733-42a9-90f8-0825f76203b8.png)

johanseto replied:
In this commit 2a4a46f ,  I can pass the pycodestyle given an extra whitespace. 
`# ################ FEAT ############` . What's your opinion about it?

felipemontoya commented on codejailservice/app.py:
```diff
@@ -56,22 +54,41 @@ def configure_codejail(app):
+
+# -----------------app configuration-------------------------
+
+app = Flask(__name__)
```
I'm not soo familiar with flask, but this general settings seem weird to me in the same app file. Can we move them to a different file that does the config and start?

johanseto replied:
In this commit e4c6724, I propose a change to move routes to their folder. This uses "blueprint" of flask.
In that case, this file(`app.py`) only configures the app. The run file is `wsgi.py`

felipemontoya commented on codejailservice/app.py:
```diff
@@ -56,22 +54,41 @@ def configure_codejail(app):
-from codejail.safe_exec import not_safe_exec as codejail_not_safe_exec
-from codejail.safe_exec import safe_exec as codejail_safe_exec
+# Lazy imports of codejail functions according(after) the configuration  app.
+code_jail_safe_exec = importlib.import_module('codejail.safe_exec')
```
This lazy import looks funky as well, maybe by moving to the other file we can make it cleaner

johanseto replied:
For this, I make a method inside the `routes/code-exec.py` call `import_code_jail_safe_exec`.  e4c6724

felipemontoya commented on requirements/dev.in:
```diff
@@ -0,0 +1,13 @@
+pytest
+pudb
+
+
```
extra line

johanseto replied:
Solved in 2a4a46f.

felipemontoya reviewed (changes requested):
Left inline comments.
We still need to find a better way to configure the app without the lazy import

johanseto committed:
clean: Clean for extra lines and style issues.

Clean according the pull request review in
https://github.com/eduNEXT/codejailservice/pull/6#discussion_r747045420.

johanseto committed:
feat: add routes config for endpoints.

Make routes  folder with each endpoint configuration.
The routes are configured with blueprint of flask.
The prevous routes were moved to and config to the routes folder.

Reconfigure app.py with the routes change.

Add `tutor.py` in gitignore.

johanseto commented:
I test the proposal for routes distribution in my local environment and codejailservice works. e4c6724
Next:

- Test in other environments.
- Setup PR and test PR.

JuanDavidBuitrago commented:
These tasks were solved in:

https://github.com/eduNEXT/codejailservice/pull/10
https://github.com/eduNEXT/codejailservice/pull/11
https://github.com/eduNEXT/codejailservice/pull/12
